packages/bvaughn-architecture-demo/components/console/hooks/useFilteredMessagesDOM.tsx

```diff
@@ -0,0 +1,151 @@
+    ).flat();
+  }, [client, eventTypesToLoad]);
+
+  const { messages } = getMessages(client, focusRange);
+
+  // Pre-filter in-focus messages by non text based search criteria.
+  const preFilteredMessages = useMemo<ProtocolMessage[]>(() => {
+    if (showErrors && showLogs && showNodeModules && showWarnings) {
+      return messages;
+    } else {
+      return messages.filter((message: ProtocolMessage) => {
+        switch (message.source) {
+          case "ConsoleAPI": {
+            switch (message.level) {
+              case "warning": {
+                if (!showWarnings) {
+                  return false;
+                }
+                break;
+              }
+              case "error": {
+                if (!showErrors) {
+                  return false;
+                }
+                break;
+              }
+              default: {
+                if (!showLogs) {
+                  return false;
+                }
+                break;
+              }
+            }
+            break;
+          }
+          case "PageError": {
+            if (!showExceptions) {
+              return false;
+            }
+            break;
+          }
+        }
+
+        // TODO This seems expensive; can we cache the message-to-node-modules relationship?
+        if (!showNodeModules) {
+          const isNodeModules = message.data.frames?.some(frame => {
+            const sourceId = frame.location?.[0].sourceId;
+            const source = sourceId ? getSourceIfAlreadyLoaded(sourceId) : null;
+            if (source) {
+              return source.url?.includes("node_modules") || source.url?.includes("unpkg.com");
+            }
+            return false;
+          });
+          if (isNodeModules) {
+            return false;
+          }
+        }
+
+        return true;
+      });
+    }
+  }, [messages, showErrors, showExceptions, showLogs, showNodeModules, showWarnings]);
+
+  const sortedLoggables = useMemo<Loggable[]>(() => {
+    const loggables: Loggable[] = [...eventTypeLogs, ...logPoints, ...preFilteredMessages];
+    return loggables.sort((a: Loggable, b: Loggable) => {
+      return getTimeForSort(a) - getTimeForSort(b);
+    });
+  }, [eventTypeLogs, logPoints, preFilteredMessages]);
+
+  useEffect(() => {
+    const needle = filterByText.toLocaleLowerCase();
+    const list = listRef.current!;
+    list.childNodes.forEach((node: ChildNode, index: number) => {
+      const element = node as HTMLElement;
+      const textContent = element.textContent?.toLocaleLowerCase();
+      const matches = textContent?.includes(needle);
+
+      // HACK Style must be compatible with the visibility check in useConsoleSearchDOM()
+      element.style.display = matches ? "inherit" : "none";
+    });
+  }, [
+    filterByText,
+    listRef,
+    showErrors,
+    showExceptions,
+    showLogs,
+    showNodeModules,
+    showWarnings,
+    sortedLoggables,
+  ]);
```

Comment:
This hook is mostly a duplicate of the previous one, except for this effect (which does the actual searching). Basically it removes a lot of the verbosity since we're only dealing with `textContent` instead of different message types.

```diff
1,7c1,6
< import { LogPointInstance } from "@bvaughn/src/contexts/LogPointsContext";
< import { isEventTypeLog, isLogPointInstance } from "@bvaughn/src/utils/console";
< import useSearch from "@bvaughn/src/hooks/useSearch";
< import type { Actions as SearchActions, State as SearchState } from "@bvaughn/src/hooks/useSearch";
< import { getCachedAnalysis } from "@bvaughn/src/suspense/AnalysisCache";
< import { Message as ProtocolMessage, Value as ProtocolValue } from "@replayio/protocol";
< import { useMemo, useState } from "react";
---
> import useSearchDOM from "@bvaughn/src/hooks/useSearchDOM";
> import type {
>   Actions as SearchActions,
>   State as SearchState,
> } from "@bvaughn/src/hooks/useSearchDOM";
> import { MutableRefObject, useMemo, useState } from "react";
9c8
< import useFilteredMessages, { Loggable } from "./useFilteredMessages";
---
> import useFilteredMessagesDOM, { Loggable } from "./useFilteredMessagesDOM";
13c12,16
< function search(query: string, loggables: Loggable[]): Loggable[] {
---
> function search(
>   query: string,
>   loggables: Loggable[],
>   listRef: MutableRefObject<HTMLElement | null>
> ): Loggable[] {
15d17
< 
17,63c19,28
<   loggables.forEach(loggable => {
<     if (isEventTypeLog(loggable)) {
<       loggable.values.some(value => {
<         // TODO Search non-primitive values (nested values) as well.
<         // Probably easier if we convert from ProtocolValue to ClientValue first.
<         if (typeof value === "string") {
<           if ((value as string).toLocaleLowerCase().includes(needle)) {
<             results.push(loggable);
<             return true;
<           }
<         } else if (typeof value?.value === "string") {
<           if (value?.value?.toLowerCase()?.includes(needle)) {
<             results.push(loggable);
<             return true;
<           }
<         }
<         return false;
<       });
<     } else if (isLogPointInstance(loggable)) {
<       const logPointInstance = loggable as LogPointInstance;
<       const analysis = getCachedAnalysis(
<         logPointInstance.point.location,
<         logPointInstance.timeStampedHitPoint,
<         logPointInstance.point.content
<       );
<       analysis?.values?.some(value => {
<         // TODO Search non-primitive values (nested values) as well.
<         // Probably easier if we convert from ProtocolValue to ClientValue first.
<         if (typeof value === "string") {
<           if ((value as string).toLocaleLowerCase().includes(needle)) {
<             results.push(loggable);
<             return true;
<           }
<         } else if (typeof value?.value === "string") {
<           if (value?.value?.toLowerCase()?.includes(needle)) {
<             results.push(loggable);
<             return true;
<           }
<         }
<         return false;
<       });
<     } else {
<       const protocolMessage = loggable as ProtocolMessage;
<       if (
<         typeof protocolMessage.text === "string" &&
<         protocolMessage.text.toLocaleLowerCase().includes(needle)
<       ) {
---
> 
>   const list = listRef.current!;
>   list.childNodes.forEach((node: ChildNode, index: number) => {
>     const element = node as HTMLElement;
> 
>     // HACK Must be compatible with the style used by useFilteredMessagesDOM()
>     if (element.style.display != "none") {
>       const textContent = element.textContent?.toLocaleLowerCase();
>       if (textContent?.includes(needle)) {
>         const loggable = loggables[index];
65,76d29
<       } else {
<         protocolMessage.argumentValues?.some((argumentValue: ProtocolValue) => {
<           // TODO Search non-primitive values (nested values) as well.
<           // Probably easier if we convert from ProtocolValue to ClientValue first.
<           if (typeof argumentValue.value === "string") {
<             if (argumentValue.value.toLocaleLowerCase().includes(needle)) {
<               results.push(loggable);
<               return true;
<             }
<           }
<           return false;
<         });
100,101c53,56
< export default function useConsoleSearch(): [State, Actions] {
<   const messages = useFilteredMessages();
---
> export default function useConsoleSearchDOM(
>   listRef: MutableRefObject<HTMLElement | null>
> ): [State, Actions] {
>   const messages = useFilteredMessagesDOM(listRef);
103c58
<   const [state, dispatch] = useSearch<Loggable>(messages, search);
---
>   const [state, dispatch] = useSearchDOM<Loggable>(messages, search, listRef);
```